extensions/flyway/runtime/src/main/java/io/quarkus/flyway/runtime/PostgreSQLSqlStatementBuilderSubstitutions.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright 2019 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.quarkus.flyway.runtime;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.flywaydb.core.internal.line.Line;
+import org.flywaydb.core.internal.sqlscript.Delimiter;
+import org.flywaydb.core.internal.sqlscript.SqlStatement;
+import org.flywaydb.core.internal.sqlscript.StandardSqlStatement;
+
+import com.oracle.svm.core.annotate.Alias;
+import com.oracle.svm.core.annotate.Substitute;
+import com.oracle.svm.core.annotate.TargetClass;
+
+/**
+ * This substitution removes de PosgreSQL COPY statement support in Flyway to allow native image compilation
+ * Without this substitution the native image generation will fail unless the PosgreSQL dependency is specified
```

Comment:
totally optional, but we could resolve this limitation:

See how we use `@TargetClass(onlyWith= ...)` in https://github.com/quarkusio/quarkus/blob/a91a034b855cae3f8d474951cb33e8d05707dc13/extensions/jaxb/runtime/src/main/java/io/quarkus/jaxb/runtime/graal/JAXBSubstitutions.java#L24-L35

I guess one could enable this subtitution only when the PostgreSQL JDBC driver is **not** available - so that the feature still works when it's there, and it gets transparently disabled when it's not there.

Reply:
incidentally, it's s/PosgreSQL/PostgreSQL/

Reply:
Done, I implement it as you said.